common/frame_session.go

```diff
@@ -78,6 +78,11 @@ type FrameSession struct {
 	// Keep a reference to the main frame span so we can end it
 	// when FrameSession.ctx is Done
 	mainFrameSpan trace.Span
+	// The initial navigation when a new page is created navigates to about:blank.
+	// We want to make sure that the the navigation span is created for this in
+	// onFrameNavigated, but subsequent calls to onFrameNavigated in the same
+	// mainframe never again create a navigation span.
+	initialNavDone bool
```

Comment:
What about `iframe`s? Do we need to consider them as they also navigate?

Reply:
There's currently a [todo for iframes](https://github.com/grafana/xk6-browser/blob/a0daaf6aba7bfd030f7decdcfbec257d96da6060/common/frame_session.go#L813). This is something we may need to look at in the future. I've not seen a use case for needing to start a navigation span for iframes as of yet.